cache/contenthash/checksum.go

```diff
@@ -856,6 +853,10 @@ func (cc *cacheContext) scanChecksum(ctx context.Context, m *mount, p string, fo
 	k := convertPathToKey(p)
 	txn := cc.tree.Txn()
 	root = txn.Root()
+
+	// only applies to Windows, it's a no-op for the rest.
+	enableProcessPrivileges()
```

Comment:
So how does this work? Does it set the privilege mode for whole application or only for running thread?

If first then is the problem that this gets called multiple times in parallel and defer from first call runs after enable from the second call?

If second then doesn't it need to be protected with https://pkg.go.dev/runtime#LockOSThread

Reply:
sure, sets the privilege for the whole process, not just the thread. `winio.RunWithPrivileges` calls `LockOSThread` though: https://github.com/microsoft/go-winio/blob/bdc6c112805ce13d7b235dc6257f258d958f5471/privilege.go#L75

Reply:
Redone, that fixes it. Thanks!